From: Takashi Iwai <tiwai@suse.de>
Subject: Fix segfault of modpost with external modules
Patch-mainline: 
References: 139359

Fix the segfault of modpost called to build out-of-tree modules.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -613,6 +613,8 @@ add_depends(struct buffer *b, struct mod
 	buf_printf(b, "__attribute__((section(\".modinfo\"))) =\n");
 	buf_printf(b, "\"depends=");
 	for (s = mod->unres; s; s = s->next) {
+		const char *r;
+
 		if (!s->module)
 			continue;
 
@@ -620,8 +622,10 @@ add_depends(struct buffer *b, struct mod
 			continue;
 
 		s->module->seen = 1;
-		buf_printf(b, "%s%s", first ? "" : ",",
-			   strrchr(s->module->name, '/') + 1);
+		r = strrchr(s->module->name, '/');
+		if (! r)
+			continue;
+		buf_printf(b, "%s%s", first ? "" : ",", r + 1);
 		first = 0;
 	}
 	buf_printf(b, "\";\n");
